hw-w2.js

```diff
@@ -0,0 +1,147 @@
+
+// with console message
+if (typeof var1 !== typeof var4);
+console.log('var1 and var4 are not the same type of variables!'); // returned: var1 and var4 are not the same type of variables!
+
+
+// *** number 9 from Javascript1 week2 ***
+
+// 9: the new value of x would be 1, because the mathematical operator '%' provides with the remainder value when the assigned value is divided by the value at the right side of '%'.
+
+// 9.1 explanatory examples:
+a = 90; //returned 90
+b = a % 9; //returned 0
+c = a % 5; //returned 0
+
+// *** number 10 from Javascript1 week2 ***
+
+// 10: yes, multiple types would be stored in an array.
+
+// 10.1
+myArray = [1, 2, 3, 'hahaha', null, [true, false]]; // returned: Array(6) [ 1, 2, 3, "hahaha", null, (2) […] ]
+
+// 10.2: comparing infinities depends on the type of value. in this case (6/0 === 10/0), it results in 'true' because === operator focuses on the types of the given values, not the values. here both of them is 'NaN'.
+
+// 10.3
+if ( 6 / 0 === 10 / 0) {
+  console.log('the provided data types of values are the same');
+} else {
+  alert('they are not the same');
+}
+
+
+
+
+
+// *** New Homework by Yash Kapila ***
+
+// *** Strings 
+
+const myString = "hello,this,is,a,difficult,to,read,sentence";
+
+console.log(myString);
+console.log(myString.length);
+console.log(myString.replace(/,/g, ' '));
+
+
+const str = 'dlroW olleH';
+function reverseString(str) {
+  const splitString = str.split('');
+  const reverseArray = splitString.reverse();
+  const joinArray = reverseArray.join('');
+  return joinArray;
+};
+console.log(reverseString(str)); // returned: Hello World
+
+// other way
+
+const reversed = str;
+Array.from(str).reverse().join('');
+console.log(Array.from(str).reverse().join('')); //returned 'Hello World'
+
+
+
+// *** Arrays 
+
+const favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+
+favoriteAnimals.push('turtle')
+
+// new array will now have 4 elements, including the value supplied with push method.
+
+console.log(favoriteAnimals);
+
+favoriteAnimals.splice(1, 0, 'snake');
+console.log(favoriteAnimals); // returned: Array(5) [ "blowfish", "snake", "capricorn", "giraffe", "turtle" ]
+
+removedAnimal = favoriteAnimals.splice(3, 1)
+// returned: Array [ "giraffe" ]
+
+console.log(favoriteAnimals); //returned: Array(4) [ "blowfish", "snake", "capricorn", "turtle" ]
+
+
+
+// *** Operators
+
+// if a == b and b == c, then a == c. this works because == operator focuses on the value, not the type. so after second statement b is equal of a and c values, meaning a is equivalent of c. but below
+
+0 == '0'; // true, coercing: JS converts each to a string when compared with == operator
+0 == [];  // true, because coercing into its numeric value so the length of [] is 0;
+"0" == [];  // false, because while "0" is falsy value, '[]' is truthy; making the result false (true + false = false)
```

Comment:
this one is tricky, javascript coerces `[]` into an empty string `""` thus this expression becomes `"0" == ""` which is `false`.